pkg/pgmodel/model/labels.go

```diff
@@ -130,10 +157,13 @@ func LabelProtosToLabels(labelPairs []prompb.Label) (*Labels, string, error) {
 	}
 	labels := GetLabels(str)
 	if labels == nil {
-		labels = new(Labels)
-		labels.str = str
-		labels.Names = make([]string, len(labelPairs))
-		labels.Values = make([]string, len(labelPairs))
+		labels = &Labels{
+			Names:    make([]string, len(labelPairs)),
+			Values:   make([]string, len(labelPairs)),
+			str:      str,
+			seriesID: unsetSeriesID,
```

Comment:
A thing for discussion. We create a new Labels object and then store in the cache with the seriesID being `-1`. Then, every time before insertion, we find out that seriesID (from table if its new) and replace -1 with the ID it deserves. Do we make sure that the cache is updated with that finding? If not, we are wasting a lot a CPU and also trips to DB for no reason.

Reply:
If we do not update the cache with new series ID, then can we do the findings of a seriesID here and insert directly in the cache with a non-default (non `-1`) seriesID, and it again saves a complete loop from the cache which we would do later on to fill the seriesID to this `-1`.

Reply:
1. Yes cache is updated, we most likely have an integration test for that
2. If you set the series id here as reported by the db, you waste the batching done currently where all the missing series id being retrieved in one query

Reply:
Yes this way is done for batching. It is updated when set because this is the same object in the cache